lib/components/app/responsive-webapp.js

```diff
@@ -268,10 +278,34 @@ const WebappWithRouter = withRouter(
  * so that Auth0 services are available everywhere.
  */
 class RouterWrapperWithAuth0 extends Component {
+  state = {
+    messages: null
+  }
+
+  componentDidMount () {
+    this.updateMessages()
+  }
+
+  componentDidUpdate (prevProps) {
+    this.updateMessages(prevProps.locale)
+  }
+
+  async updateMessages (previousLocale) {
+    // If the locale changes (or no prevProps are provided), update the messages
+    // for that locale.
+    const {customMessages, locale} = this.props
+    if (locale && (!previousLocale || locale !== previousLocale)) {
+      const messages = await loadLocaleData(locale, customMessages)
+      this.setState({messages})
+    }
+  }
```

Comment:
I know @landonreed wants to avoid putting the messages within the redux state, but I really don't understand why. This is just adding unnecessary boilerplate since the state of this component is derived exclusively and unconditionally from the redux state. This is [a react anti-pattern](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-unconditionally-copying-props-to-state).

Reply:
Does that advice apply to state derived using async calls?

Reply:
I was resistant at first because it seemed somewhat unnecessary since the messages are not really shared state (they're only used by this component), but I'm happy to be overridden if we think the code would be cleaner by stashing them in the redux store/state.

Reply:
Even though the value of the messages is only used within this specific component, this value is influenced and can change as a result of an event originating from another component. Regarding Binh's question, my objection doesn't have to do with whether this is async. If some kind of component-specific value uses an async function to obtain it's value, but that initiation event of the call to that async function comes from that same component and the state is not shared in any other parallel component hierarchy then it might be fine to keep it in that component. However, I find that putting logic into actions/reducers makes the code more organized and can keep components decluttered.

Reply:
I have created an action that loads the localized strings when the locale is set (https://github.com/opentripplanner/otp-react-redux/pull/375/commits/6ea4204258ce54cf8da5a953043aeaffefdbef67).